src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -508,13 +508,14 @@ impl Qualif for IsNotConst {
     }
 }
 
-// Refers to temporaries which cannot be promoted as
-// promote_consts decided they weren't simple enough.
-// FIXME(oli-obk,eddyb): Remove this flag entirely and
-// solely process this information via `IsNotConst`.
-struct IsNotPromotable;
+/// Refers to temporaries which cannot be promoted *implicitly*.
+/// Explicit promotion e.g. for constant arguments declared via `rustc_args_required_const` or by
```

Comment:
This still doesn't read like a complete sentence to me, did you mean to add an "is" before "e.g."?

Reply:
Hm... that sentence sounded better when I wrote it XD I must have lost my train of thought while writing it. Better now?